core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1760,7 +1780,42 @@ private DynamicFilter getDynamicFilter(
             }
 
             log.debug("[TableScan] Dynamic filters: %s", dynamicFilters);
-            return context.getDynamicFiltersCollector().createDynamicFilter(dynamicFilters, tableScanNode.getAssignments(), context.getTypes());
+            DynamicFilter localDynamicFilter = context.getLocalDynamicFiltersCollector().createDynamicFilter(dynamicFilters, tableScanNode.getAssignments(), context.getTypes());
+
+            DynamicFilter coordinatorDynamicFilter = context.createCoordinatorDynamicFilter(dynamicFilters, tableScanNode, metadata, typeOperators);
+            return new DynamicFilter() {
```

Comment:
Why do we have to merge local and coordinator dynamic filters?
We can just use `getLocalDynamicFiltersCollector`

Reply:
ya, the merging is unnecessary now (it's left over from some previous iteration).
i've simplified this now, one reason to keep `LocalDynamicFiltersCollector#register` is if we want to implement a flag for this change, then if the flag is off we need to avoid registering non-local DFs in LocalExecutionPlanner.
i can remove it if we don't want to add the flag.